File: application-tracker/application-tracker.Server/Helper/UserHelper.cs
using System.Threading.Tasks;
using System.Collections.Generic;
using application_tracker.Server.Models;
using Microsoft.EntityFrameworkCore;

namespace application_tracker.Server.Helper;
public static class UserHelper
{
    public static async Task PopulateNewUserAsync(ApplicationUser user, ApplicationDbContext dbContext)
    {
        if (dbContext.Entry(user).State == EntityState.Detached)
        {
            dbContext.Attach(user);
        }
        ApplicationFolder defaultFolder = new ApplicationFolder
        {
            Name = "Favorites",
            Owner = user,
            OwnerId = user.Id,
        };
        ApplicationFolder defaultFolder2 = new ApplicationFolder
        {
            Name = "Archived",
            Owner = user,
            OwnerId = user.Id,
        };
        JobApplication defaultJob = new JobApplication
        {
            CompanyName = "FDM Group",
            Position = "Social Media Intern.",
            Notes = "",
            Status = ApplicationStatus.Wishlist,
            Owner = user,
            OwnerId = user.Id,
            JobPostingUrl = "",
        };
        JobApplication defaultJob2 = new JobApplication
        {
            CompanyName = "Google",
            Position = "Software Engineer",
            Notes = "Applied through referral",
            Status = ApplicationStatus.Applied,
            Owner = user,
            OwnerId = user.Id,
            JobPostingUrl = "https://careers.google.com/jobs/results/1234567890-software-engineer/",
        };
        JobApplication defaultJob3 = new JobApplication
        {
            CompanyName = "Microsoft",
            Position = "Data Scientist",
            Notes = "Interview scheduled for next week",
            Status = ApplicationStatus.Rejected,
            Owner = user,
            OwnerId = user.Id,
            JobPostingUrl = "https://careers.microsoft.com/jobs/results/0987654321-data-scientist/",
        };

        defaultFolder.Applications.Add(defaultJob);
        defaultFolder.Applications.Add(defaultJob2);
        defaultFolder2.Applications.Add(defaultJob2);
        defaultFolder2.Applications.Add(defaultJob3);
        using var transaction = await dbContext.Database.BeginTransactionAsync();
        try
        {
            await dbContext.ApplicationFolders.AddRangeAsync(defaultFolder, defaultFolder2);
            await dbContext.JobApplications.AddRangeAsync(defaultJob, defaultJob2, defaultJob3);
            await dbContext.SaveChangesAsync();
            await transaction.CommitAsync();
        }
        catch
        {
            await transaction.RollbackAsync();
            throw;
        }
    }
}

File: application-tracker/server/src/lib/user-helper.ts
import { db } from "../db";
import {
  applicationFolders,
  applicationFolderJobApplication,
  jobApplications,
} from "../db/schema";

export async function populateNewUser(userId: string) {
  await db.transaction(async (tx) => {
    const [favorites] = await tx
      .insert(applicationFolders)
      .values({ name: "Favorites", ownerId: userId })
      .returning({ id: applicationFolders.id });

    const [archived] = await tx
      .insert(applicationFolders)
      .values({ name: "Archived", ownerId: userId })
      .returning({ id: applicationFolders.id });

    const [fdm] = await tx
      .insert(jobApplications)
      .values({
        companyName: "FDM Group",
        position: "Social Media Intern.",
        notes: "",
        status: 0,
        ownerId: userId,
        jobPostingUrl: "",
      })
      .returning({ id: jobApplications.id });

    const [google] = await tx
      .insert(jobApplications)
      .values({
        companyName: "Google",
        position: "Software Engineer",
        notes: "Applied through referral",
        status: 1,
        ownerId: userId,
        jobPostingUrl: "https://careers.google.com/jobs/results/1234567890-software-engineer/",
      })
      .returning({ id: jobApplications.id });

    const [microsoft] = await tx
      .insert(jobApplications)
      .values({
        companyName: "Microsoft",
        position: "Data Scientist",
        notes: "Interview scheduled for next week",
        status: 4,
        ownerId: userId,
        jobPostingUrl: "https://careers.microsoft.com/jobs/results/0987654321-data-scientist/",
      })
      .returning({ id: jobApplications.id });

    await tx.insert(applicationFolderJobApplication).values([
      { folderId: favorites.id, applicationId: fdm.id },
      { folderId: favorites.id, applicationId: google.id },
      { folderId: archived.id, applicationId: google.id },
      { folderId: archived.id, applicationId: microsoft.id },
    ]);
  });
}
